### PR TITLE
[examples] unit test for run_bart_sum

### PR DESCRIPTION
**Pull request by sshleifer**
- add lightning to `examples/requirements.txt`
- first lightning unittest!

### CODE REVIEW

File: examples/transformer_base.py [hunk 1]
```diff
@@ -102,8 +102,8 @@ def optimizer_step(self, epoch, batch_idx, optimizer, optimizer_idx, second_orde
         self.lr_scheduler.step()
 
     def get_tqdm_dict(self):
-        tqdm_dict = {"loss": "{:.3f}".format(self.trainer.avg_loss), "lr": self.lr_scheduler.get_last_lr()[-1]}
-
+        avg_loss = getattr(self.trainer, "avg_loss", 0.0)
```
Comment (sshleifer) [inline examples/transformer_base.py]: `self.trainer.avg_loss` was raising `AttributeError`

File: examples/summarization/bart/run_bart_sum.py [hunk 2]
```diff
@@ -8,7 +8,12 @@
 from torch.utils.data import DataLoader
 
 from transformer_base import BaseTransformer, add_generic_args, generic_train, get_linear_schedule_with_warmup
-from utils import SummarizationDataset
+
+
+try:
```
Comment (sshleifer) [inline examples/summarization/bart/run_bart_sum.py:13]: to support being from from pwd or examples/

File: examples/summarization/bart/run_train_tiny.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Script for verifying that run_bart_sum can be invoked from its directory
```
Comment (sshleifer) [inline examples/summarization/bart/run_train_tiny.sh:1]: Runs same logic as train on a tiny dataset

File: examples/requirements.txt [hunk 4]
```diff
@@ -5,4 +5,5 @@ seqeval
 psutil
 sacrebleu
 rouge-score
-tensorflow_datasets
\ No newline at end of file
+tensorflow_datasets
+git+http://github.com/PyTorchLightning/pytorch-lightning/@master
```
Comment (julien-c) [inline examples/requirements.txt:9]: is there a released version that we can use?

File: examples/summarization/bart/test_bart_examples.py [hunk 5]
```diff
@@ -31,3 +77,21 @@ def test_bart_cnn_cli(self):
...
+        args_d.update(
+            data_dir=make_test_data_dir(),
+            model_type="bart",
+            train_batch_size=2,
+            eval_batch_size=2,
+            n_gpu=n_gpu,
+            output_dir="dringus",
+        )
+        args = argparse.Namespace(**args_d)
+        os.makedirs(args.output_dir, exist_ok=False)
```
Comment (julien-c) [inline examples/summarization/bart/test_bart_examples.py:96]: you can drop the `exist_ok ` as False it the default

**Comment by julien-c** [comment]
LGTM from a superficial glance

File: examples/summarization/bart/run_bart_sum.py [hunk 6]
```diff
@@ -8,7 +8,12 @@
...
+    from .utils import SummarizationDataset
+except ImportError:
+    from utils import SummarizationDataset
```
Comment (julien-c) [inline examples/summarization/bart/run_bart_sum.py:16]: I asked @aaugustin if there's an easy way to not resort to this kind of hack

**Comment by sshleifer** [comment]
Planning on merging this April 15 at 7pm EST barring objections.

**Pull request merged**